### ISSUE TITLE
Loss Functions

### ISSUE THREAD

**Issue by BeeGass** [BeeGass/Your-ML-Library#6]
* Regression Loss Functions
  * [x] Mean Squared Error Loss (MSE)
  * [x] Mean Absolute Error Loss (MAE)
  * [x] Mean Absolute Percentage Error (MAPE)
  * [x] Mean Squared Logarithmic Error Loss (MSLE)
  * [x] Mean Bias Error (MBE)
  * [x] Squared Error Loss
  * [x] Cosine Similarity Loss
  * [x] Binary Cross Entropy Loss
  * [x] Cross Entropy Loss
  * [ ] Logarithmic Hyperbolic Cosine Loss
  * [x] Huber Loss
  * [ ] Poisson Loss

* Classification Losses
  * Hinge Losses for "maximum-margin" Classification
    * [ ] Hinge Loss
    * [ ] Squared Hinge Loss
    * [ ] Categorical Hinge Loss

  * Binary Classification Loss Functions
    * [ ] Binary Cross-Entropy
    * [ ] Binary Logistic Loss
    * [ ] Sigmoid_binary_cross_entropy

  * Multi-class Classification Loss Functions
    * [ ] Softmax Cross_Entropy 
    * [ ] Multi-class Cross Entropy Loss
    * [ ] Sparse Multi-class Cross-Entropy
    * [ ] Multi-class Logistic Loss
    * [ ] Multi-class Sparsemax Loss
    * [ ] Negative Log Likelihood
    * [ ] Kullback Leibler Divergence Loss
    * [ ] SVM Loss
    
* Other Loss Functions That Need To Be Looked Into
  * [ ] Connectionist Temporal Classification loss
  * [ ] Negative log likelihood loss with Poisson distribution
  * [ ] Gaussian negative log likelihood loss.
  * [ ] Binary Cross Entropy With Logits Loss
  * [ ] Margin Ranking Loss
  * [ ] Hinge Embedding Loss
  * [ ] Multi-Label Margin Loss
  * [ ] Smooth L1 Loss
  * [ ] Soft Margin Loss
  * [ ] Multi-Label Soft Margin Loss
  * [ ] Cosine Embedding Loss
  * [ ] Multi-Margin Loss
  * [ ] Triplet Margin Loss
  * [ ] Triplet Margin With Distance Loss
